Add Flatpak workflow procedures for Satellite (#3624)
theforeman/foreman-documentation: guides/common/modules/proc_enabling-the-flatpak-remote.adoc

[id="enabling-the-flatpak-remote"]
= Enabling the Flatpak remote

This procedure configures and manages Flatpak repositories by using Hammer CLI.

.Prerequisites
 * Your {Project} account has a role that grants the permissions `view_flatpak_remotes`, `create_flatpak_remotes`, `edit_flatpak_remotes`, and `destroy_flatpak_remotes`. 

* Set up Flatpak. 
For more information, see {RHELDocsBaseURL}9/html/administering_the_system_using_the_gnome_desktop_environment/assembly_installing-applications-using-flatpak_administering-the-system-using-the-gnome-desktop-environment#setting-up-flatpak_assembly_installing-applications-using-flatpak[Setting up Flatpak].

* Flatpak applications rely on Flatpak runtimes. 
For example, `rhel9/firefox-flatpak` depends on `rhel9/flatpak-runtime`. 
* Ensure that runtime repositories are available to clients alongside application repositories for installations to work.

.CLI procedure
. Enable a Flatpak remote on the {ProjectServer} by using standalone {EL} systems or {ProjectName}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> $ hammer flatpak-remote create --name=_My_Flatpak_Remote_Name_ --url=$url \

Reviewer: ```suggestion
$ hammer flatpak-remote create --name=_My_Flatpak_Remote_Name_ --url=_My_Flatpak_Remote_URL_ \
```

--organization=your_organization
----
+
You can include authentication details by using the options `--username=_My_User_Name_ --token=_My_Token_`.
ifdef::satellite[]
+
You can generate a {Team} official token at link:https://access.redhat.com/terms-based-registry[Registry Service Accounts].
endif::[]

. Update the Flatpak remote with authentication credentials:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ hammer flatpak-remote update --username=_My_User_Name_ --token=_My_Token_
----
. Optional: List and view information about the Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ hammer flatpak-remote list
$ hammer flatpak-remote info
----
. Scan the Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ hammer flatpak-remote scan --id=_My_ID_
----
. List repositories in the Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ hammer flatpak-remote repository list --flatpak-remote-id=_My_ID_
----
. Mirror a Flatpak remote repository to a {Project} product:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> $ hammer flatpak-remote remote-repository mirror --id=_My_Remote_Repo_id_ --product-id=$satellite_product

Reviewer: I'm just guessing:

```suggestion
$ hammer flatpak-remote remote-repository mirror --id=_My_Remote_Repo_id_ --product-id=_{Project}_Product_ID_
```

[id="enabling-the-flatpak-remote"]
= Enabling the Flatpak remote

> This Hammer CLI procedure enables and manages Flatpak repositories.

Reviewer: Quoting https://redhat-documentation.github.io/supplementary-style-guide/#shortdesc:

> * Avoid self-referential language, such as "This topic covers…​" or "Use this procedure to…​".
> [...]
> * Use customer-centric language, such as "You can…​ by…​" or "To…​, configure…​".

.Prerequisites
* *view_flatpak_remotes*: The user can view Flatpak remotes.
* *create_flatpak_remotes*: The user can create new Flatpak remotes.
* *edit_flatpak_remotes*: Grants the user permission to change existing Flatpak remotes.

> * *destroy_flatpak_remotes*: The user can delete Flatpak remotes.

Reviewer: This currently misses an introduction: What are you listing? The answer is permissions and it would be good to state that explicitly. I searched for other modules that include user permissions as prerequisites and this is a nice example: https://github.com/theforeman/foreman-documentation/blob/cf0c81548b54ee2480395210e96fcd9d2889f838/guides/common/modules/proc_customizing-host-registration-by-using-snippets.adoc?plain=1#L17
Note the markup and the fact that we don't typically explain what each permission does. (It's actually pretty self-explanatory.)

Set up Flatpak. 
For more information, see link:https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/8/html/using_the_desktop_environment_in_rhel_8/assembly_installing-applications-using-flatpak_using-the-desktop-environment-in-rhel-8#setting-up-flatpak_assembly_installing-applications-using-flatpak[Setting up Flatpak].

Ensure the assigned user role includes these permissions to manage Flatpak remotes without admin privileges.

Flatpak applications rely on Flatpak runtimes. 
For example, `rhel9/firefox-flatpak` depends on `rhel9/flatpak-runtime`. 

> Ensure that runtime repositories are available to clients alongside application repositories for installations to work.

Reviewer: It seems these are missing bullet points.

.CLI procedure
. Enable a Flatpak remote on the {ProjectServer} by using standalone {EL} systems or {ProjectName}:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> $ hammer flatpak-remote create --name=_My_Name_ --url=$url \

Reviewer: A more descriptive replaceable would seem like a good choice here. Just a guess:
```suggestion
$ hammer flatpak-remote create --name=_My_Flatpak_Remote_Name_ --url=$url \
```

--organization=your_organization
----
+

> You can include authentication details during creation by using the options `--username=$username --token=$token`.

Reviewer: ```suggestion
You can include authentication details by using the options `--username=_My_User_Name_ --token=_My_Token_`.
```

> +
> You can generate a {Team} official token at link:https://access.redhat.com/terms-based-registry[Registry Service Accounts].

Reviewer: ```suggestion
ifdef::satellite[]
+
You can generate a {Team} official token at link:https://access.redhat.com/terms-based-registry[Registry Service Accounts].
endif::[]
```

. Update the Flatpak remote with authentication credentials:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----

> $ hammer flatpak-remote update --username=_My_User_Name_ --token=_My_Token_

Reviewer: This is an alternative to line 30, right? Generally speaking, we are encouraged to choose one way of doing things rather than presenting users with multiple alternatives. Which one of these two ways of providing credentials should we recommend to users in this procedure? (They can find all the other options in hammer CLI help or the Hammer CLI guide.)

----
. List and view information about the Flatpak remote:
+
[options="nowrap", subs="+quotes,verbatim,attributes"]
----
$ hammer flatpak-remote list
$ hammer flatpak-remote info

> ----

Reviewer: This is not really part of 'enabling Flatpak remote'. You can either drop the whole step (users will be able to find this in hammer --help or in the Hammer guide or the release notes, actually), or you can at least mark it as an optional step.

One reason why I'm suggesting this is that it's good to keep in mind that we don't want our procedures to be too long. Long procedures are more difficult to read and can even look intimidating at first glance. If you're writing a procedure on 'enabling a Flatpak remote', then it's good to include steps relevant to that particular user goal rather than try to document all the commands that have been added.
